mailpoet/lib/Segments/WP.php

```
@@ -182,10 +182,35 @@ private function handleCreatingOrUpdatingSubscriber(string $currentFilter, \WP_U
-    } catch (\Exception $e) {
-      return; // fails silently as this was the behavior of this methods before the Doctrine refactor.
+    // When updating an existing subscriber's email, remove any other subscriber
+    // that already holds the new email to avoid a unique constraint violation.
+    // This can happen when a WP user registers with email A, checks out with
+    // email B (creating a second subscriber), then changes their account email
+    // from A to B.
+    // Uses a DBAL-level transaction (not EntityManager::wrapInTransaction) to
+    // keep the delete and update atomic without risking a permanently closed
+    // EntityManager or premature postRemove lifecycle events.
+    if ($subscriber !== null && $subscriber->getEmail() !== $data['email']) {
+      $this->databaseConnection->beginTransaction();
```

PHPStan (2.1.46)

[error] Call to method beginTransaction() on an unknown class MailPoetVendor\Doctrine\DBAL\Connection.
Learn more at https://phpstan.org/user-guide/discovering-symbols

(class.notFound)